Dan-Mark-EnemyAbstract

bakerdg11 opened the pull request:
Created EnemyAbstract script with basic templates. We commented out a pathfinding method as we may build on that. We included a basic Shoot script, however we wanted to confirm who was responsible for Shoot and Projectile script for PlayerController and Enemy scripts.

Wanted to include in the review everyone who is working on Enemy Scripts to determine if we're missing anything or want anything adjusted. 

Ziyad, also went to remove Log scripts but figured we'd leave it in to allow testing until a PlayerController is merged.

bakerdg11 committed:
Dan-Mark-EnemyAbstract

Created EnemyAbstract script with basic templates.
We commented out a pathfinding method as we may build on that.
We included a basic Shoot script, however we wanted to confirm who was responsible for Shoot and Projectile script for PlayerController and Enemy scripts.

MV1090 reviewed (commented):
I would also like add that all the logic inside right now isn't final, all of this will probably change when we have a better understand of how things will function. If anyone has any questions on how to use this script with their specific enemy class we can have a discussion on this with in the meeting on Monday, or just reach out on discord.

SolomanDavis commented on Assets/_Scripts/EnemyAbstract.cs:
```diff
@@ -0,0 +1,79 @@
+
+        StartCoroutine(ShootCoroutine());
+
+        EnemyHealth = 2;
```
This should probably be a `[SerializeField]` so we can specify it in the editor

MV1090 replied:
Is this related to EnemyHealth??

SolomanDavis reviewed (changes requested):
just 1 change

SolomanDavis commented on Assets/_Scripts/EnemyAbstract.cs:
```diff
@@ -0,0 +1,79 @@
+using Unity.VisualScripting;
+using UnityEngine;
+
+public abstract class EnemyAbstract : MonoBehaviour
```
The name should just be `Enemy`

bakerdg11 committed:
Dan Mark Enemy and Projectile Script

Updated Enemy script with the suggestions from Ziyad.
Updated Shoot function, perameters Speed will equal speed of projectile and Offset will determine how close to the player the enemy shoots, based on players position when projectile is spawned. Added enemy death method to be called by enemy animation event.
Added Projectile script. Once PlayerController is added, we will add adjust the velocity of the projectile. Has OnCollision when player is hit to start IsDead animation on Player

bakerdg11 commented:
Updated Enemy script with the suggestions from Ziyad.
Updated Shoot function, perameters Speed will equal speed of projectile and Offset will determine how close to the player the enemy shoots, based on players position when projectile is spawned. Added enemy death method to be called by enemy animation event.
Added Projectile script. Once PlayerController is added, we will add adjust the velocity of the projectile. Has OnCollision when player is hit to start IsDead animation on Player

SolomanDavis commented on Assets/_Scripts/Enemy/Enemy.cs:
```diff
@@ -0,0 +1,84 @@
+        bc = GetComponent<BoxCollider2D>();
+        anim = GetComponent<Animator>();
+
+        EnemyHealth = 2;
```
Note that this will set whatever is configured in the unity editor for `EnemyHealth` to `2` whenever the game starts

MV1090 replied:
Yeah that's correct, we figured as the most times an enemy can be hit is twice (for the heavy enemy's) they didn't need more then 2 health. smaller enemies will take 2 damage while the heavies will only take 1, there for it takes two shot to kill them. We can always but this into a method where each enemy can set its own health but we didn't think it was necessary as for the reasons above. Let us know if you want us to approach this differently and we can change it.

SolomanDavis replied:
ohh I see - so when a player projectile hits light or medium enemies, those enemies need to take 2 damage, and when it hits a heavy enemy, it needs to take 1 damage. That works fine but my concern here is that this assignment is overriding what's being set in the unity editor. If for some reason we want to test and change `EnemyHealth` to 10 in unity, It'll just get set back to 2 here.

I agree with taking 2 damage but I'm thinking that this `EnemyHealth = 2` isn't needed here and can be set as a default where the field is defined and explicitly in the unity editor.

MV1090 replied:
I see what your saying, my bad I didn't realize this was set in the start, I thought we set it where it was declared at the top. We should change this.

SolomanDavis commented on Assets/_Scripts/Enemy/EnemyProjectile.cs:
```diff
@@ -0,0 +1,44 @@
+    Rigidbody2D rb;
+
+    [HideInInspector] public int speed;
+    [HideInInspector] public int offset;
```
Why are these two variables hidden in the inspector?

MV1090 replied:
We chose to hide these two in the inspector because they have to be public as they are called from another class but we don't want their values changed with in the inspector as they are set within the enemy class.

EstelleSchoeman reviewed (approved):
If you resolve Ziyad's comments, I believe it's ready to merge.
Good work.

Commit:
Dan - Mark

Removed the EnemyHealth from start as per the request, this value can be in the inspector.

Commit:
Merge branch 'master' into Dan-Mark-AbstractEnemy

# Conflicts:
#	Assets/_Scripts/Enemy.meta
#	Assets/_Scripts/Managers/CanvasManager.cs
